templates/terraform/expand_property_method.erb

```diff
@@ -59,11 +60,11 @@ func expand<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d T
 
 <%# Generate expanders for flattened objects %>
 <%   elsif property.flatten_object -%>
-func expand<%= prefix -%><%= titlelize_property(property) -%>(d TerraformResourceData, config *Config) (interface{}, error) {
+func expand<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d TerraformResourceData, config *Config) (interface{}, error) {
```

Comment:
If we're never going to use the `v` value why do we set one at all? The parameter being called `param1` everywhere make it feel that we're really not sure what to do with it.

Reply:
1. Once I get `flatten_object` to be tolerant of un-flattened objects partway through the stack this will need to be passing an arbitrary `v` through the stack and the only thing that will be hard-coded to `nil` will be the top level call in the Create/Update method. Inside this method it would check if `v` is nil + if the current object is flattened to determine whether to pass `v` through or to call `.Get`.
2. Even if we don't get 1. working this keeps the method signature the same, which IMO makes for much cleaner templating logic at the cost of a bit less clean downstream. Since the templating logic is getting increasingly complicated I think it's a worth while trade off.

For now I can just plumb `v` through instead of explicitly passing `nil` if you think that's more clear?

Reply:
Not sure I follow 1.

Having a dynamic value like `param1` in our templates feels worse to me than outputting a different method signature based on a conditional. It's harder to reason about what code will be outputted as a result of this change.

Reply:
Also 2 is probably just me bikeshedding so don't take that part _too_ seriously.

Reply:
oops in my original response 1 & 2 were in response to "why set `v` at all"

`param1` just felt slightly cleaner than inlining and that felt a bit better than if/else pasta.

Follow up on why set `v`: If we end up supporting an api structure of `one.two.three.four` where `one` and `three` are flattened but `two` we will need to pass the object from the expandTwo to expandFour since the Terraform schema would be `two.four`.
Also there are lots of examples in our generated code of method params that aren't consumed.

Reply:
> Also there are lots of examples in our generated code of method params that aren't consumed.

Yeah that's why I realised I was just bikeshedding on 2.

---

> Follow up on why set v

I may have forgotten something you told me before, but I'm still not sure I understand the mechanics of why we need to pass this.

---

This context isn't clear as a reader of the code. I know that it's hard to include complex context in this kind of code but I'd like to surface something to make this a little more clear.

It sounds like `param1` is some kind of ancestor/parent, so we could look at naming it as such. We could also explain *why* it will be `nil`, and what the implications of that are in comment form.